tfhe/src/integer/server_key/comparator.rs

```diff
@@ -1689,4 +2268,502 @@ mod tests {
+                assert_eq!(a, clear_0);
+            }
+
+            let decrypted_result: U256 = cks.decrypt_radix(&encrypted_result);
+
+            let expected_result = clear_fn(clear_0, clear_1);
+            assert_eq!(decrypted_result, expected_result);
+        }
+    }
+
+    /// This macro generates the tests for a given scalar comparison fn
+    ///
+    /// All our scalar comparison function have 3 variants:
+    /// - unchecked_scalar_$comparison_name_parallelized
+    /// - smart_scalar_$comparison_name_parallelized
+    /// - scalar_$comparison_name_parallelized
+    ///
+    /// So, for example, for the `gt` comparison fn,
+    /// this macro will generate the tests for the 3 variants described above
+    macro_rules! define_scalar_comparison_test_functions {
+        ($comparison_name:ident) => {
+            paste::paste!{
+
+                fn [<unchecked_scalar_ $comparison_name _parallelized_256_bits>](params:  crate::shortint::ClassicPBSParameters) {
+                    let num_tests = 1;
+                    test_unchecked_scalar_function(
+                        params,
+                        num_tests,
+                        |comparator, lhs, rhs| comparator.[<unchecked_scalar_ $comparison_name _parallelized>](lhs, rhs),
+                        |lhs, rhs| U256::from(<U256>::$comparison_name(&lhs, &rhs) as u128),
+                    )
+                }
+
+                fn [<smart_scalar_ $comparison_name _parallelized_256_bits>](params:  crate::shortint::ClassicPBSParameters) {
+                    let num_tests = 1;
+                    test_smart_scalar_function(
+                        params,
+                        num_tests,
+                        |comparator, lhs, rhs| comparator.[<smart_scalar_ $comparison_name _parallelized>](lhs, rhs),
+                        |lhs, rhs| U256::from(<U256>::$comparison_name(&lhs, &rhs) as u128),
+                    )
+                }
+
+                fn [<scalar_ $comparison_name _parallelized_256_bits>](params:  crate::shortint::ClassicPBSParameters) {
+                    let num_tests = 1;
+                    test_default_scalar_function(
+                        params,
+                        num_tests,
+                        |comparator, lhs, rhs| comparator.[<scalar_ $comparison_name _parallelized>](lhs, rhs),
+                        |lhs, rhs| U256::from(<U256>::$comparison_name(&lhs, &rhs) as u128),
+                    )
+                }
+
+                create_parametrized_test!([<unchecked_scalar_ $comparison_name _parallelized_256_bits>]
+                {
+                    PARAM_MESSAGE_2_CARRY_2,
+                    PARAM_MESSAGE_3_CARRY_3,
+                    PARAM_MESSAGE_4_CARRY_4
+                });
+
+                create_parametrized_test!([<smart_scalar_ $comparison_name _parallelized_256_bits>]
+                {
+                    PARAM_MESSAGE_2_CARRY_2,
+                    // We don't use PARAM_MESSAGE_3_CARRY_3,
```

Comment:
3_3 still broken in that case ?

Reply:
See https://github.com/zama-ai/tfhe-rs/pull/336#discussion_r1224556779